### PR TITLE
Created Ldap connection functionality with spring security

### PR DESCRIPTION
**Pull request by tim-boerner**

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/enums/PasswordEncryption.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.data.database.helper.enums;
+
+/**
+ * Enum for type of password encryption, each one with integer value for database.
+ */
+public enum PasswordEncryption {
+    SHA(0,"SHA"),
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/enums/PasswordEncryption.java]: Missing space between method parameters.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/enums/PasswordEncryption.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    MD5(1,"MD5");
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/helper/enums/PasswordEncryption.java]: Missing space between method parameters.

File: Kitodo/src/main/java/org/kitodo/security/DynamicAuthenticationProvider.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+     */
+    public static DynamicAuthenticationProvider getInstance() {
+        return instance;
+    }
+
+    /**
+     * This method reads local config and sets authentication flag.
+     */
+    public void readLocalConfig() {
+        ldapAuthentication = ConfigCore.getBooleanParameter("ldap_use",false);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/security/DynamicAuthenticationProvider.java]: Missing space between method parameters.

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 4]
```diff
@@ -819,4 +828,19 @@ private FileManagementInterface getFileManagementModule() {
...
+     * Deletes the slash as first character from an uri object.
+     * @param uri
+     *          The uri object.
+     * @return
+     *          The new uri object without first slash.
+     */
+    public URI deleteFirstSlashFromPath(URI uri) {
+        String uriString = uri.getPath();
+        if (uriString.startsWith("/")) {
+            uriString = uriString.replaceFirst("/","");
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: Missing space between method parameters.

File: Kitodo/src/test/java/org/kitodo/services/data/LdapGroupServiceIT.java [hunk 5]
```diff
@@ -41,5 +43,7 @@ public void shouldFindLdapGroup() throws Exception {
         LdapGroup ldapGroup = ldapGroupService.getById(1);
         boolean condition = ldapGroup.getTitle().equals("LG") && ldapGroup.getDisplayName().equals("Name");
         assertTrue("LDAP group was not found in database!", condition);
+
+        assertEquals("Title of Ldap server is not matching","FirstLdapServer",ldapGroup.getLdapServer().getTitle());
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/data/LdapGroupServiceIT.java]: Missing space between method parameters.

File: Kitodo/src/test/java/org/kitodo/services/data/UserServiceIT.java [hunk 6]
```diff
@@ -507,4 +508,16 @@ public void shouldNotSaveUserWithSameLogin() throws DataException {
         exception.expect(DataException.class);
         userService.save(newUser);
     }
+
+    @Test
+    public void shouldGetLdapServerOfUser() throws DAOException {
+        User user = userService.getById(2);
+        assertEquals("LdapServer title is incorrect!","FirstLdapServer",user.getLdapGroup().getLdapServer().getTitle());
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/data/UserServiceIT.java]: Missing space between method parameters.

File: Kitodo/src/test/java/org/kitodo/services/data/UserServiceIT.java [hunk 7]
```diff
@@ -507,4 +508,16 @@ public void shouldNotSaveUserWithSameLogin() throws DataException {
...
+    }
+
+    @Test
+    public void shouldGetUserByLdapLogin() throws DAOException {
+        User user = userService.getByLdapLogin("kowalLDP");
+        assertEquals("User surname is incorrect!","Kowalski",user.getSurname());
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/data/UserServiceIT.java]: Missing space between method parameters.

File: Kitodo/src/test/java/org/kitodo/services/file/FileServiceTest.java [hunk 8]
```diff
@@ -531,4 +531,11 @@ public void testCreateBackupFile() throws IOException {
         Assert.assertFalse(fileService.fileExist(URI.create("2/meta.xml.3")));
     }
 
+    @Test
+    public void testDeleteFirstSlashFromPath() {
+        URI uri = URI.create("/test/test");
+        URI actualUri = fileService.deleteFirstSlashFromPath(uri);
+        Assert.assertEquals("Paths of Uri did not match","test/test",actualUri.getPath());
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/file/FileServiceTest.java]: Missing space between method parameters.

File: Kitodo/src/main/java/de/sub/goobi/forms/LoginForm.java [hunk 9]
```diff
@@ -67,9 +66,8 @@ public String PasswortAendernSpeichern() {
             Helper.setFehlerMeldung(Helper.getTranslation("passwordsDontMatch"));
         } else {
             try {
-                /* wenn alles korrekt, dann jetzt speichern */
-                Ldap myLdap = new Ldap();
-                myLdap.changeUserPassword(this.myBenutzer, this.passwortAendernAlt, this.passwortAendernNeu1);
+                /* wenn alles korrekt, dann jetzt speichern */;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/LoginForm.java]: Why you added a `; ` at end of line?

**Comment by tim-boerner** [comment]
Requested changes done. Thanks for Review!

**Pull request merged**